metron-analytics/metron-maas-service/pom.xml

```diff
@@ -59,6 +59,23 @@
       <artifactId>hadoop-yarn-server-common</artifactId>
       <version>${hadoop.version}</version>
       <scope>provided</scope>
+      <exclusions>
```

Comment:
One thing we might consider with these bigger dep exclusions (and this is just an idea, entirely up to you) is using wildcards. 

```
        <dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-common</artifactId>
            <version>2.7.1</version>
                <exclusions>
                    <exclusion>
                        <groupId>org.apache.curator</groupId>
                        <artifactId>*</artifactId>
                    </exclusion>
                </exclusions>
        </dependency>
```

I found this results in all curator deps from hadoop-common being excluded whereas previously I had the following:

```
grep curator /tmp/deptree.txt
[INFO]    |  \- org.apache.curator:curator-framework:jar:2.7.1:compile
[INFO]    +- org.apache.curator:curator-client:jar:2.7.1:compile
[INFO]    +- org.apache.curator:curator-recipes:jar:2.7.1:compile
```

I'm not sure if this is too big a net to cast or not. The dependency plugin doesn't appear to do globs, only full wildcard matching at the artifactId or groupId level. e.g. "curator*" will not work in the artifactId, unfortunately. Just an idea.

This is the Maven feature - https://issues.apache.org/jira/browse/MNG-2315

Reply:
That's a nice feature.  I'd prefer to use. Unfortunately, I tried it like your example and I don't seem to work as I'd expect.  When I use a wildcard like your example, I still get old 2.12.0 dependencies getting pulled in.